deploy/deployment-manager/deploy.sh

```diff
@@ -86,25 +91,33 @@ is_role_not_assigned() {
   fi
 }
 
+# Set environment variables with the name and number of your project.
+export PROJECT_NAME=$(gcloud config get-value core/project)
+export PROJECT_NUMBER=$(gcloud projects list --filter="${PROJECT_NAME}" --format="value(PROJECT_NUMBER)")
+
+# Fail fast if any of the required environment variables are not provided
+check_env_not_provided "FLEET_URL"
+check_env_not_provided "ENROLLMENT_TOKEN"
+check_env_not_provided "STACK_VERSION"
+
+configure_scope
+
 # Enable the Google Cloud APIs needed for misconfiguration scanning
 run_command "gcloud services enable iam.googleapis.com deploymentmanager.googleapis.com compute.googleapis.com cloudresourcemanager.googleapis.com cloudasset.googleapis.com"
 
 ## If needed add the required role to apply the templates to the project using the deployment manager
 ADD_ROLE=false
 if is_role_not_assigned; then
-  run_command "gcloud projects add-iam-policy-binding ${PROJECT_NAME} --member=serviceAccount:${PROJECT_NUMBER}@cloudservices.gserviceaccount.com --role=${ROLE}"
+  run_command "gcloud ${SCOPE} add-iam-policy-binding ${PARENT_ID} --member=serviceAccount:${PROJECT_NUMBER}@cloudservices.gserviceaccount.com --role=${ROLE}"
   ADD_ROLE=true
 fi
 
 # Apply the deployment manager templates
 run_command "gcloud deployment-manager deployments create --automatic-rollback-on-error ${DEPLOYMENT_NAME} --project ${PROJECT_NAME} \
     --template compute_engine.py \
-    --properties elasticAgentVersion:${STACK_VERSION},fleetUrl:${FLEET_URL},enrollmentToken:${ENROLLMENT_TOKEN},allowSSH:${ALLOW_SSH},zone:${ZONE},elasticArtifactServer:${ELASTIC_ARTIFACT_SERVER} \
-    --labels $DEPLOYMENT_LABELS"
+    --properties elasticAgentVersion:${STACK_VERSION},fleetUrl:${FLEET_URL},enrollmentToken:${ENROLLMENT_TOKEN},allowSSH:${ALLOW_SSH},zone:${ZONE},elasticArtifactServer:${ELASTIC_ARTIFACT_SERVER},scope:${SCOPE},parentId:${PARENT_ID}"
 
 ## Remove the role required to deploy the DM templates
 if [ "$ADD_ROLE" = "true" ]; then
-  run_command "gcloud projects remove-iam-policy-binding ${PROJECT_NAME} --member=serviceAccount:${PROJECT_NUMBER}@cloudservices.gserviceaccount.com --role=${ROLE}"
+  run_command "gcloud ${SCOPE} remove-iam-policy-binding ${PARENT_ID} --member=serviceAccount:${PROJECT_NUMBER}@cloudservices.gserviceaccount.com --role=${ROLE}"
 fi
-
-echo -e "\nDeployment completed successfully, elastic agent enrollment might take a few minutes."
```

Comment:
This message is incorrect.
The deployment can fail, but we use the `--automatic-rollback-on-error` flag, and as a result of a successful rollback the command returns 0 as an exit code.